pkg/datapath/linux/config/config.go

```diff
@@ -440,12 +445,27 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 
 		if option.Config.EnableIPv4 {
 			nodePortIPv4Addrs := node.GetNodePortIPv4AddrsWithDevices()
-			ipv4 := byteorder.NetIPv4ToHost32(nodePortIPv4Addrs[directRoutingIface])
+			ip, ok := nodePortIPv4Addrs[directRoutingIface]
+			if !ok {
+				log.WithFields(logrus.Fields{
+					"nodePortIPv4Addrs":  nodePortIPv4Addrs,
+					"directRoutingIface": directRoutingIface,
+				}).Fatal("NodePort enabled but direct routing device's IP not found")
+			}
+
+			ipv4 := byteorder.NetIPv4ToHost32(ip)
 			cDefinesMap["IPV4_DIRECT_ROUTING"] = fmt.Sprintf("%d", ipv4)
 		}
 
 		if option.Config.EnableIPv6 {
-			directRoutingIPv6 := node.GetNodePortIPv6AddrsWithDevices()[directRoutingIface]
+			directRoutingIPv6, ok := node.GetNodePortIPv6AddrsWithDevices()[directRoutingIface]
+			if !ok {
+				log.WithFields(logrus.Fields{
+					"directRoutingIPv6":  directRoutingIPv6,
```

Comment:
This field will always be `nil`, perhaps it should just be removed.

Reply:
Hi @ti-mo , thanks for you comments!
But I didn't get the idea here, do you mean `directRoutingIPv6` will always be `nil`, why?

Reply:
See https://golang.org/ref/spec#Index_expressions:
> if the map is nil or does not contain such an entry, a[x] is the zero value for the element type of M

As a result, when `ok` is false, `directRoutingIPv6` will be an empty `net.IP`.

Reply:
Refreshed my golang skills, thanks :) @ti-mo